Solutions/ISC Bind/Parsers/ISCBind.txt

```diff
@@ -0,0 +1,86 @@
+// Title:           ISC Bind
+// Author:          Microsoft
+// Version:         1.0
+// Last Updated:    09/14/2022
+// Comment:         Inital Release
+//  
+// DESCRIPTION:
+// This parser takes raw ISC Bind logs from a Syslog stream and parses the logs into a normalized schema.
+//
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as ISCBind
+// 4. Kusto Functions can typically take up to 15 minutes to activate. You can then use Function Alias for other queries.
+//
+// REFERENCES: 
+// Using functions in Azure monitor log queries: https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+
+let request = Syslog 
+| where SyslogMessage has_all ("client", "query:") and SyslogMessage !has "response:"
+    // -- trim_end needed to remove optional repetative IP address at the end.
+| extend SyslogMessage = trim_end (@' \([\d\.]+\)', SyslogMessage)
+| extend SyslogMessage = replace_string (SyslogMessage, '\\ ', '@@@')
+| parse SyslogMessage with
+        * "client " * " "
+        SrcIpAddr:string "#" 
+        SrcPortNumber:int *
+        "query: "
+        DnsQuery:string " "
+        DnsQueryClassName:string " "
+        DnsQueryTypeName:string " "
+        DnsFlags:string
+| extend DnsQuery = replace_string (DnsQuery, '@@@', ' ')
+| extend 
+        EventSubType = "request",
+        DnsResponseCodeName = "NA"
+| project-away SyslogMessage,ProcessName,ProcessID,Facility,SeverityLevel,HostName;
+
+
+let requestcache = Syslog 
+| where SyslogMessage has_all ("client", "query (cache)") and SyslogMessage !has "response:"
+    // -- trim_end needed to remove optional repetative IP address at the end.
+| extend SyslogMessage = trim_end (@' \([\d\.]+\)', SyslogMessage)
```

Comment:
We can remove this as request cache does not have the server IP. Also, the next line with replace.